modules/jdbc-test/src/test/java/org/testcontainers/junit/AbstractContainerDatabaseTest.java

```diff
@@ -26,7 +31,16 @@ DataSource getDataSource(JdbcDatabaseContainer container) {
         hikariConfig.setJdbcUrl(container.getJdbcUrl());
         hikariConfig.setUsername(container.getUsername());
         hikariConfig.setPassword(container.getPassword());
+        hikariConfig.setDriverClassName(container.getDriverClassName());
+
+        final HikariDataSource dataSource = new HikariDataSource(hikariConfig);
+        datasourcesForCleanup.add(dataSource);
+
+        return dataSource;
+    }
 
-        return new HikariDataSource(hikariConfig);
+    @After
+    public void teardown() {
+        datasourcesForCleanup.forEach(HikariDataSource::close);
```

Comment:
I was noticing a lot of `Creating connection` logs from the Clickhouse driver during unrelated tests, long after the Clickhouse tests completed.

I realised that the Hikari connection pools were not being closed after use, and were hanging around trying to reconnect to containers that were long dead.

I've added this cleanup code to the parent test class to ensure that the connection pools are always closed, regardless of how they are used. We could alternatively use try-with-resources, but we'd have to make sure we use it in all the tests that are based on this abstract class.